### PR TITLE
Fix the CloudFlare handling in the RequestBuffer

### PR DESCRIPTION
**Pull request by ArsenArsen**
### Prerequisites
* [x] I have read and understood the [contribution guidelines](CONTRIBUTING.md)
* [x] This pull request follows the code style of the project
* [x] I have tested this feature thoroughly

**Issues Fixed:** A CloudFlare caused 502 would usualy cancel the request, now it will tryAgain() the request.

### Changes Proposed in this Pull Request
* Add a CloudFlare Exception
* Make the above mentioned CloudFlare exception invoke a return tryAgain() in the RequestBuffer class.

### CODE REVIEW

File: src/main/java/sx/blah/discord/util/CloudFlareException.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package sx.blah.discord.util;
+
+/**
+ * In case of a CloudFlare interruption this exception gets thrown
+ * <br>
+ * Created by Arsen on 22.9.16..
```
Comment (theIglooo) [inline src/main/java/sx/blah/discord/util/CloudFlareException.java]: You should remove the default javadoc message

File: src/main/java/sx/blah/discord/util/LogMarkers.java [hunk 2]
```diff
@@ -59,7 +59,12 @@
 	 * The marker for websocket keepalive actions.
 	 * It is a child of {@link #WEBSOCKET} and {@link #VOICE_WEBSOCKET}.
 	 */
-	KEEPALIVE(WEBSOCKET, VOICE_WEBSOCKET);
+	KEEPALIVE(WEBSOCKET, VOICE_WEBSOCKET),
+	/**
+	 * The marker for all websocket-related logging.
+	 * It is a child of {@link #API}.
+	 */
+	CLOUDFLARE(WEBSOCKET);
```
Comment (theIglooo) [inline src/main/java/sx/blah/discord/util/LogMarkers.java]: This isn't necessary for just one log message, use the provided UTIL marker. The javadoc on this is also wrong

Comment (ArsenArsen) [inline src/main/java/sx/blah/discord/util/LogMarkers.java]: Okay, will fix now.

File: src/main/java/sx/blah/discord/handle/impl/obj/Channel.java [hunk 3]
```diff
@@ -486,7 +486,8 @@ private void overridePermissions(String type, String id, EnumSet<Permissions> to
 	}
 
 	@Override
-	public List<IInvite> getInvites() throws DiscordException, RateLimitException {
```
Comment (GrandPanda) [inline src/main/java/sx/blah/discord/handle/impl/obj/Channel.java]: Changes in a PR should be focues on one feature/bug fix.

File: src/main/java/sx/blah/discord/api/internal/Requests.java [hunk 4]
```diff
@@ -122,8 +123,9 @@ private Request(Class<? extends HttpUriRequest> clazz, IDiscordClient client) {
 		 *
 		 * @throws RateLimitException
 		 * @throws DiscordException
+		 * @throws CloudFlareException
 		 */
-		public String makeRequest(String url, BasicNameValuePair... headers) throws RateLimitException, DiscordException {
+		public String makeRequest(String url, BasicNameValuePair... headers) throws RateLimitException, DiscordException, CloudFlareException {
```
Comment (GrandPanda) [inline src/main/java/sx/blah/discord/api/internal/Requests.java]: It is so obvious that you did not test this change in the *slightest*. This change is **extremely** breaking. You'd added a checked exception to one of the most commonly used methods in the entire library.... Open up *any* accessor of this method and you'll be shown a wall of errors for the uncaught exception.....

**Pull request closed without merging**